zigpy/quirks/v2/__init__.py

```diff
@@ -243,6 +252,46 @@ def __call__(self, device: CustomDeviceV2) -> None:
                 )
 
 
+@attrs.define(frozen=True, kw_only=True, repr=True)
+class AddsEndpointMetadata:
+    """Adds metadata for adding an endpoint to a device."""
+
+    endpoint_id: int = attrs.field()
+    profile_id: int = attrs.field()
+    device_type: int = attrs.field()
+
+    def __call__(self, device: CustomDeviceV2) -> None:
+        """Process the add."""
+        if self.endpoint_id not in device.endpoints:
+            device.add_endpoint(self.endpoint_id)
+            setattr(device.endpoints[self.endpoint_id], SIG_EP_PROFILE, self.profile_id)
```

Comment:
Since this is working with the `Endpoint` object directly, I think `ep = device.add_endpoint(...)` and `ep.profile_id = ...` would be clearer.

Reply:
Ah yeah, that's what I initially had, but mypy complained, as it's of type `ZDO | Endpoint` and `ZDO` doesn't have `profile_id` and `device_type`. I'll take another look at this.

Reply:
I think `ep = device.endpoints[self.endpoint_id]` would do it but `ep = device.add_endpoint(...)` should always be a `Endpoint` type, not `Endpoint | ZDO`.

Reply:
Yeah, I missed that part of your message. I initially had `ep = device.endpoints[self.endpoint_id]` which (obviously) didn't work. Fixed with https://github.com/zigpy/zigpy/pull/1524/commits/db5759b53eb454787c0105dc02bd9f1158a23f7e now.